src/test/java/com/google/jenkins/plugins/computeengine/ComputeEngineCloudIT.java

```diff
@@ -284,8 +315,67 @@ public void testWorkerFailed() throws Exception {
         assertEquals(logs(), true, logs().contains("WARNING"));
     }
 
-    private static InstanceConfiguration validInstanceConfiguration1(String numExecutors) {
-        return instanceConfiguration(DEB_JAVA_STARTUP_SCRIPT, numExecutors, LABEL);
+    @Test(timeout = 300000)
+    public void testTemplate() throws Exception {
+        ComputeEngineCloud cloud = (ComputeEngineCloud) r.jenkins.clouds.get(0);
+        cloud.addConfiguration(validInstanceConfigurationWithTemplate(TEMPLATE));
+       
+        InstanceTemplate instanceTemplate = new InstanceTemplate();
+        instanceTemplate.setName(nameFromSelfLink(TEMPLATE));
+        InstanceProperties instanceProperties = new InstanceProperties();
+        instanceProperties.setMachineType(nameFromSelfLink(MACHINE_TYPE));
+        instanceProperties.setLabels(ImmutableMap.of("test-label", "label-value"));
+        AttachedDisk boot = new AttachedDisk();
+        boot.setBoot(true);
+        boot.setAutoDelete(BOOT_DISK_AUTODELETE);
+        boot.setInitializeParams(new AttachedDiskInitializeParams()
+                .setDiskSizeGb(Long.parseLong(BOOT_DISK_SIZE_GB_STR))
+                .setDiskType(nameFromSelfLink(BOOT_DISK_TYPE))
+                .setSourceImage(BOOT_DISK_IMAGE_NAME)
+        );
+        instanceProperties.setDisks(of(boot));
+        instanceProperties.setTags(new Tags().setItems(of(NETWORK_TAGS)));
+        instanceProperties.setMetadata(new Metadata().setItems(of(new Metadata.Items()
+                .set(METADATA_LINUX_STARTUP_SCRIPT_KEY, DEB_JAVA_STARTUP_SCRIPT))));
+        instanceProperties.setNetworkInterfaces(of(new NetworkInterface()
+                .setName(NETWORK_NAME)
+                .setAccessConfigs(of(new AccessConfig()
+                        .setType(NAT_TYPE)
+                        .setName(NAT_NAME)
+                ))
+        ));
+        instanceTemplate.setProperties(instanceProperties);
+        client.getCompute().instanceTemplates().insert(cloud.projectId, instanceTemplate).execute();
```

Comment:
Same comment as above, would prefer to maintain the encapsulation of the ComputeClient class.